### DISCUSSION TITLE
Package subpath './build/src/requireHook' is not defined

### DISCUSSION THREAD

**Discussion by farooq-zaman-82** [adonisjs/core#4768] [Help] [answered]
Hi All,

I am new to @adonisjs and trying to setup an existing project. After installing npm packages I run following command

`node ace migration:run`

but it results in following error

```
 Error: Package subpath './build/src/requireHook' is not defined by "exports" in
   /Users/fzaman/webinar/amazon-ivs-ugc-sample/web/node_modules/@adonisjs/assembler/package.json
 

   ⁃ resolveFileName
     /Users/fzaman/webinar/amazon-ivs-ugc-sample/web/node_modules/resolve-from/index.js:29
   ⁃ resolveFrom
     /Users/fzaman/webinar/amazon-ivs-ugc-sample/web/node_modules/resolve-from/index.js:43
   ⁃ module.exports
     /Users/fzaman/webinar/amazon-ivs-ugc-sample/web/node_modules/resolve-from/index.js:46
```

Following is the exports section in @adonisjs/assember/package.json file

```
  "exports": {
    ".": "./build/index.js",
    "./code_transformer": "./build/src/code_transformer/main.js",
    "./types": "./build/src/types.js"
  },
```

How do I solve this issue? Any help in this regard is highly appreciated.

Following is my development setup:

Node: v23.0.0 (I tried multiple versions)
NPM: v10.9.0
@adonisjs/assember: 7.7.0 (Also tried latest version.

**Comment by thetutlage** [comment]
Can you please share your application as a Github repo for me to reproduce the issue?

**Reply by farooq-zaman-82** [reply]
I am trying following public repo. Apparently it's not installing the assembler package:

[https://github.com/aws-samples/amazon-ivs-ugc-sample/tree/main/web](url)

`node ace --version` command gives me following output:

```
╭────────────────────────────────────────────╮
│    node ace --version                      │
│────────────────────────────────────────────│
│                                            │
│    App version: 1.0.0                      │
│    Framework version: 5.9.0                │
│    Assembler version: Not Installed        │
│                                            │
╰────────────────────────────────────────────╯
```

**Reply by RomainLanz** [reply, marked as answer]
It looks like you’re using AdonisJS 5, but the assembler version you installed is compatible with AdonisJS 6.

To keep using AdonisJS 5, please follow the recommended versions here: https://v6-migration.adonisjs.com/guides/stick-to-v5

**Reply by farooq-zaman-82** [reply]
This fixed the issue. Thanks a lot.
